homeassistant/components/ecovacs/event.py

```diff
@@ -58,3 +78,41 @@ async def on_event(event: ReportStatsEvent) -> None:
             self.async_write_ha_state()
 
         self._subscribe(self._capability.event, on_event)
+
+
+class EcovacsLastPositionEventEntity(
+    EcovacsEntity[VacuumCapabilities, CapabilityEvent[PositionsEvent]],
+    EventEntity,
+):
+    """Ecovacs last job event entity."""
+
+    entity_description = EventEntityDescription(
+        key="last_position",
+        translation_key="last_position",
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        event_types=[POSITIONS_UPDATED_EVENT],
+    )
+
+    def __init__(self, device: Device[VacuumCapabilities]) -> None:
+        """Initialize entity."""
+        super().__init__(device, device.capabilities.map.position)
+
+    async def async_added_to_hass(self) -> None:
+        """Set up the event listeners now that hass is ready."""
+        await super().async_added_to_hass()
+
+        async def on_event(event: PositionsEvent) -> None:
+            """Handle event."""
+            event_data: dict[str, dict] = {
+                position.type.name.lower(): {
```

Comment:
The positions in `PositionsEvent` are on purpose a list as someone could have multiple charges, which this entity currently does not support.

Please don't make any changes to this PR until I have discussed it internally and come back to you. So we don't waste your time ;)